### ISSUE TITLE
is there any training code?

### ISSUE THREAD

**Issue by JuHyung-Son** [V-Sense/DeepNormals#4]
I cannot find training code

**Comment by MatisHudon** [comment]
Hello, 

Thanks for your interest. You didn't find it because we did not provide such code as it is usually really dependent on the data used (tfrecords, images ...). Although we provided the complete model architecture in the model.py file, therefore it should be quite straightforward to re-train the model from there should you need to.
If you have any question I would be happy to help.

**Comment by JuHyung-Son** [comment]
Thanks for the early comment.

The reason you split the input into many tiles is that the input image has a high resolution?

Do you think the tile and multi-scale representation works for the input image with the size 512 x 512 x 3 ?

**Comment by MatisHudon** [comment]
Yes this is  to avoid down-scaling the input and therefore obtain high resolution output. This should normally work with input size of 512x512x3.
Please don't hesitate if you have any problem.

**Comment by JuHyung-Son** [comment]
But, I wonder that you can just make the network bigger so that the network has high-resolution input and the same size output.

**Comment by JuHyung-Son** [comment]
Also, model need mask of a image when inferencing. But there's no mask data on dataset. Should I make it respectively?

**Comment by MatisHudon** [comment]
This method is compatible with any input size and will output a normal map of the same size as the input. Keep in mind that the number of weights to train will increase with the size of the network therefore at some point you might reach some hardware/memory limitations. Also to train on full images directly (without tiling) you might require a bigger dataset.
All the details are in the paper linked where we fully explain why we made this choice (of tiling) and also compare it to a fully convolutional approach.

The masks can be easily generated from the normal map images.
